### ISSUE TITLE
IBM comment on 1.4.14 Content on Hover or Focus

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag21#439]
IBM encourages clarifying language in the Understanding document that discourages the practice of content appearing only on focus or mouse hover. As discussed in the issue thread, this SC is designed to reduce the problems resulting from this practice, and it should be made clear that the existence of the SC is not itself a recommendation to expand use of such mechanisms.

**Comment by steverep** [comment]
@mbgower, can you clarify your main concern.  I assume you are referring primarily to hover because there is no such concept on touch screens?  Or the distracting nature when this stuff covers up other content?

**Comment by mbgower** [comment]
Our main concern is that author's should not be encouraged to create content that appears on hover (any more than they should be encouraged to create content that appears on focus). As you've noted, it poses problems for anyone reviewing content on mobile devices (which at the moment do not have a convention for mimicking 'on hover' behaviour). Beyond that, it has the following concerns:

- accidental activiation
similar to the reason that On Focus rules out changes of context. It is easy to accidentally trigger something on hover. The user should be able to anticipate changes in context.

- scale
depending on the implementation of an On Hover action, I think they can easily constitute a change of context. As it stands, there is no language preventing this, especially if this SC does not address it in the Understanding doc. But for the same reason you don't want Change of Context with on input or on focus, you don't want it on hover.

In the interest of eating my company's dogfood here, I'll reference a crap implementation on [IBM's current home page](https://www.ibm.com/ca-en/).
![IBM landing page](https://user-images.githubusercontent.com/14298245/32676618-2bb6994c-c610-11e7-9a39-61a178ea6bea.png)

Hovering over the top navigation items labelled Products or Services, causes an overlay to appear which covers 75% of the viewport for me.
![IBM landing page with megamenu overlay exposed by hoving on Products text](https://user-images.githubusercontent.com/14298245/32676874-1055d8ce-c611-11e7-88e1-fad5a437b40f.png)

The same overlay, which I think pretty clearly amounts to a Change of Context, is launched by an activation by keyboard. It does not appear on focus, and for that we can be truly grateful. Note that the same overlay is opened On Click. In other words, the hover effect acts as a method of opening the megamenu.

I'll note that more normal menus are displayed on hover for the items labelled Industries and Developers.
![Landing page with Industries submenu exposed](https://user-images.githubusercontent.com/14298245/32676926-49cd4bfa-c611-11e7-96c8-f3fd0084084a.png)

 I'd still call that a crappy implementation -- they are menus and they should be triggered by an activation action -- and we need langauge discouraging their use -- but at least the small submenus displayed do not amount to a change of context.

In summary, maybe there are two things I'm ID'ing here. First, we should have explicit language saying "On hover isn't recommended, but if you are going to do it, here are things you must address as an author". There are other SCs and techniques that say such things. Second, I think we need clear language -- maybe in the normative text itself, that causing a Change of Context using this SC is a failure.

**Comment by steverep** [comment]
I can craft something for the other concerns, but the change in context is not appropriate for this SC, which is all about perception rather than predictable behavior.  It would be very simple to include this in SC 3.2.1 On Focus, if not banned from changing it of course :).

**Comment by awkawk** [comment]
We will ensure that this topic is addressed in the Understanding docs.
